### PR TITLE
MDEXP-421 - Export job triggered with invalid data stays in status New (and does not fail)

### PR DESCRIPTION
**Pull request by obozhko-folio**
[MDEXP-421](https://issues.folio.org/browse/MDEXP-421) - Export job triggered with invalid data stays in status New (and does not fail)

## Purpose
If the file that is used to trigger has a supported file extension but stores invalid data, the export should fail.

## Approach
Add error handling.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/service/file/reader/LocalStorageCsvSourceReader.java [hunk 1]
```diff
@@ -63,7 +62,14 @@ public void init(FileDefinition fileDefinition, ErrorLogService errorLogService,
 
   @Override
   public boolean hasNext() {
-    return iterator.hasNext();
+    boolean result;
```
Comment (alekGbuz) [inline src/main/java/org/folio/service/file/reader/LocalStorageCsvSourceReader.java:65]: Maybe it is better to set up result as false by default and log error message for exception.

File: src/main/java/org/folio/service/file/storage/LocalFileSystemStorage.java [hunk 2]
```diff
@@ -128,6 +128,9 @@ private String getFilePath(FileDefinition fileDefinition) {
 
   @Override
   public boolean isFileExist(String path) {
+    if (path == null) {
+      return false;
+    }
     return fileSystem.existsBlocking(path);
```
Comment (alekGbuz) [inline src/main/java/org/folio/service/file/storage/LocalFileSystemStorage.java:134]: Possible to return checking at same line path !=null and fileSystem.existsBlocking(path)

File: src/main/java/org/folio/service/file/storage/LocalFileSystemStorage.java [hunk 3]
```diff
@@ -128,6 +128,9 @@ private String getFilePath(FileDefinition fileDefinition) {
 
   @Override
   public boolean isFileExist(String path) {
+    if (path == null) {
```
Comment (alekGbuz) [inline src/main/java/org/folio/service/file/storage/LocalFileSystemStorage.java:131]: Path empty string checking maybe useful here.

File: src/main/java/org/folio/service/file/reader/LocalStorageCsvSourceReader.java (hunk 1 above)
Comment (obozhko-folio) [reply src/main/java/org/folio/service/file/reader/LocalStorageCsvSourceReader.java:65]: Yes, agree.

File: src/test/java/org/folio/service/file/storage/FileStorageUnitTest.java [hunk 4]
```diff
@@ -46,6 +46,11 @@ void shouldSaveFileDataBlocking() throws IOException {
     String savedFileContent = new String(Files.readAllBytes(savedFile.toPath()));
     assertEquals(fileContent, savedFileContent);
     assertTrue(fileStorage.isFileExist(savedFileDefinition.getSourcePath()));
+    // case when path is null
+    String sourcePath = savedFileDefinition.getSourcePath();
+    savedFileDefinition.setSourcePath(null);
```
Comment (alekGbuz) [inline src/test/java/org/folio/service/file/storage/FileStorageUnitTest.java:51]: Checking if sourcePath is empty string  could be useful here.

File: src/main/java/org/folio/service/manager/export/strategy/InstanceExportStrategyImpl.java [hunk 5]
```diff
@@ -109,7 +116,17 @@ private void generateRecordsOnTheFly(ExportPayload exportPayload, List<String> i
     exportPayload.setExportedRecordsNumber(srsLoadResult.getUnderlyingMarcRecords().size() - failedSrsRecords + mappedMarcRecords.size() - failedRecordsCount);
     exportPayload.setFailedRecordsNumber(identifiers.size() - exportPayload.getExportedRecordsNumber());
     if (exportPayload.isLast()) {
-      exportService.postExport(fileExportDefinition, params.getTenantId());
+      try {
+        exportService.postExport(fileExportDefinition, params.getTenantId());
+      } catch (ServiceException exc) {
+        jobExecutionService.getById(exportPayload.getJobExecutionId(), params.getTenantId()).onSuccess(res -> {
+          Optional<JsonObject> optionalUser = usersClient.getById(fileExportDefinition.getMetadata().getCreatedByUserId(),
```
Comment (alekGbuz) [inline src/main/java/org/folio/service/manager/export/strategy/InstanceExportStrategyImpl.java:123]: Checking that user is present could be useful here.

File: src/test/java/org/folio/service/file/storage/FileStorageUnitTest.java (hunk 4 above)
Comment (obozhko-folio) [reply src/test/java/org/folio/service/file/storage/FileStorageUnitTest.java:51]: Added, thank you.

File: src/main/java/org/folio/service/manager/export/strategy/InstanceExportStrategyImpl.java (hunk 5 above)
Comment (obozhko-folio) [reply src/main/java/org/folio/service/manager/export/strategy/InstanceExportStrategyImpl.java:123]: Alright, fixed.

File: src/main/java/org/folio/service/file/storage/LocalFileSystemStorage.java (hunk 2 above)
Comment (obozhko-folio) [reply src/main/java/org/folio/service/file/storage/LocalFileSystemStorage.java]: Agree, thank you.

File: src/main/java/org/folio/service/file/storage/LocalFileSystemStorage.java (hunk 3 above)
Comment (obozhko-folio) [reply src/main/java/org/folio/service/file/storage/LocalFileSystemStorage.java]: Fixed, thank you.

**Review by alekGbuz — APPROVED** [review]

**Review by siarhei-charniak — APPROVED** [review]

**Pull request merged**